datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs

```diff
@@ -272,79 +272,129 @@ impl ApproxPercentileAccumulator {
     ) -> Result<Vec<OrderedFloat<f64>>> {
         match values.data_type() {
             DataType::Float64 => {
-                let array = values.as_any().downcast_ref::<Float64Array>().unwrap();
+                let array = values.as_any().downcast_ref::<Float64Array>().ok_or(
+                    DataFusionError::Internal(format!(
+                        "A Float64 type array, {:?} is somehow not a Float64Array",
+                        values
```

Comment:
I assume this won't print out 1GB of values in binary?

Reply:
This won't. The following function is usually used when implementing `Debug`.
https://github.com/apache/arrow-rs/blob/master/arrow/src/array/array.rs#L719-L757